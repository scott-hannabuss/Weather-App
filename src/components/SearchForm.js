import React from "react";
import PropTypes from "prop-types";
import "../styles/SearchForm.css";

const SearchForm = ({ searchText, setSearchText, onSubmit }) => {
  const handleInputChange = (event) => setSearchText(event.target.value);

  return (
    <div className="search-form">
      <h1 className="header">React Weather App</h1>
      <input
        className="input"
        type="text"
        placeholder="Enter UK City"
        onChange={handleInputChange}
        value={searchText}
      />
      <button className="search-button" type="submit" onClick={onSubmit}>
        Search
      </button>
    </div>
  );
};

export default SearchForm;

SearchForm.propTypes = {
  searchText: PropTypes.string.isRequired,
  setSearchText: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};
