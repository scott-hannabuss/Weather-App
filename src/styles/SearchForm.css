@import url("https://fonts.googleapis.com/css2?family=Chango&family=Lora:ital@1&family=Roboto+Mono:wght@300&display=swap");
@import url("https://fonts.googleapis.com/css2?family=Playfair+Display:wght@600&display=swap");

.header {
  justify-content: center;
  align-content: center;
  text-align: center;
  font-family: "Playfair Display", serif;
  width: 100%;
  font-size: 300%;
  color: white;
  margin-top: 3%;
  /* transform: scaleX(1); */
}

.input {
  margin: auto;
  margin-top: 3%;
  justify-content: center;
  align-content: center;
  text-align: center;
  display: grid;
  width: 40%;
  padding: 0.5%;
  color: black;
  border-radius: 25px;
  font-family: "Playfair Display", serif;
}

.search-button {
  margin: auto;
  color: black;
  justify-content: center;
  align-content: center;
  display: grid;
  padding: 0.5%;
  margin-top: 3%;
  width: 20%;
  border-radius: 25px;
  font-family: "Playfair Display", serif;
}

@media only screen and (min-width: 751px) and (max-width: 1025px) {
  .header {
    margin-top: 7.5%;
  }
  .input {
    margin-top: 7.5%;
  }
  .search-button {
    margin-top: 7.5%;
  }
}

@media only screen and (max-width: 750px) {
  .header {
    margin-top: 10%;
    margin-right: 5%;
    margin-left: 5%;
    width: auto;
  }
  .input {
    margin-top: 10%;
    width: 70%;
  }
  .search-button {
    margin-top: 10%;
    width: 35%;
  }
}
