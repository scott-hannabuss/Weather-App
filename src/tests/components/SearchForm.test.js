import React from "react";
import { render } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";

describe("SearchForm", () => {
  const validProps = {
    onInputChange: () => {},
    onSubmit: () => {},
    searchInput: "Liverpool",
  };

  it("renders correctly", () => {
    const { asFragment } = render(<SearchForm {...validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
