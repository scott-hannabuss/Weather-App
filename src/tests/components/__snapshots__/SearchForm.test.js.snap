// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <h1
      class="header"
    >
      React Weather App
    </h1>
    <input
      class="input"
      placeholder="Enter UK City"
      type="text"
      value="Liverpool"
    />
    <button
      class="search-button"
      type="submit"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
